### PR TITLE
AudioCore: Take suspend lock when stalling the running process.

### PR DESCRIPTION
**Pull request by bunnei**
There is a lock returned by StallProcesses() to prevent callers to Un/StallProcesses from stomping on each other. This fixes assert spam in Pokemon Scarlet/Violet where this code path tends to kick in frequently.

Moves IsPaused() to use an atomic instead of taking the suspend lock, as AudioCore calls this when it may be holding the suspend lock already. This means IsPaused may be slightly stale, but that should be acceptable for how it is used.

### CODE REVIEW

**Review by FernandoS27 — APPROVED** [review]

File: src/audio_core/sink/sink_stream.h [hunk 1]
```diff
@@ -241,7 +247,7 @@ class SinkStream {
     /// Set via IAudioDevice service calls
     f32 device_volume{1.0f};
     /// True if coretiming has been stalled
-    bool stalled{false};
+    std::unique_lock<std::mutex> stalled_lock;
```
Comment (liamwhite) [inline src/audio_core/sink/sink_stream.h:250]:
```suggestion
    std::mutex stall_guard;
    std::unique_lock<std::mutex> stalled_lock;
```

File: src/audio_core/sink/sink_stream.cpp [hunk 2]
```diff
@@ -266,19 +266,18 @@ void SinkStream::ProcessAudioOutAndRender(std::span<s16> output_buffer, std::siz
 }
 
 void SinkStream::Stall() {
-    if (stalled) {
+    if (IsStalled()) {
         return;
     }
```
Comment (liamwhite) [inline src/audio_core/sink/sink_stream.cpp:271]:
```suggestion
    std::scoped_lock lk{stall_guard};
    if (stalled_lock) {
        return;
    }
```

File: src/audio_core/sink/sink_stream.cpp [hunk 3]
```diff
@@ -266,19 +266,18 @@ void SinkStream::ProcessAudioOutAndRender(std::span<s16> output_buffer, std::siz
...
-    stalled = true;
-    system.StallProcesses();
+    stalled_lock = system.StallProcesses();
 }
 
 void SinkStream::Unstall() {
-    if (!stalled) {
+    if (!IsStalled()) {
         return;
     }
```
Comment (liamwhite) [inline src/audio_core/sink/sink_stream.cpp:278]:
```suggestion
    std::scoped_lock lk{stall_guard};
    if (!stalled_lock) {
        return;
    }
```

**Review by liamwhite — COMMENTED** [review]
Most of the changes in this PR are likely extraneous -- what I suggested should be all that's really needed.

**Review by liamwhite — APPROVED** [review]

**Pull request merged**